app/controllers/api/v1/npq/users_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module NPQ
+      class UsersController < Api::ApiController
+        include ApiTokenAuthenticatable
+
+        def create
+          creation_response = ::NPQ::Users::FindOrCreateBy.new(params: find_or_create_params).call
+
+          if creation_response.user.present?
+            hash = Api::V1::UserSerializer.new(creation_response.user).serializable_hash
+            status_code = creation_response.new_user ? :created : :ok
+
+            render json: hash, status: status_code
+          else
+            errors = [
+              creation_response.errors,
+              creation_response.error,
+            ].flatten.compact
+
+            hash = {
+              errors: errors.map do |error|
+                {
+                  status: "401",
+                  title: error.attribute,
+                  detail: error.message,
+                }
+              end,
+            }
+
+            render json: hash, status: :unauthorized
```

Comment:
Having started on the historical syncing work, and adding in an update API. I realise that unauthorized is probably the wrong status code.